tests/test_custom_checkers/test_possibly_undefined_checker.py

```diff
@@ -469,3 +470,149 @@ def g(a: int) -> int:
 
         with self.assertNoMessages():
             self.checker.visit_functiondef(func_node)
+
+
+class TestPossiblyUndefinedCheckerZ3Option(pylint.testutils.CheckerTestCase):
+    CHECKER_CLASS = PossiblyUndefinedChecker
+
+    def setup_method(self) -> None:
+        super().setup_method()
+        self.linter.config.z3 = True
+
+    def test_z3_unfeasible_branch(self) -> None:
+        src = """
+        def func(x: int) -> int:
+            '''
+            Preconditions:
+                - x > 5
+            '''
+            if x > 3:
+                a = 10
+            else:
+                pass
+            return a
+        """
+        z3v = Z3Visitor()
+        mod = z3v.visitor.visit(astroid.parse(src))
+        mod.accept(CFGVisitor())
+        func_node = mod.body[0]
+        *_, name_node = mod.nodes_of_class(nodes.Name)
+
+        with self.assertNoMessages():
+            self.checker.visit_functiondef(func_node)
```

Comment:
Don't call two different `visit` methods here. If you need both to activate for the message to be triggered, visit the top-level `Module` node instead